### PR TITLE
Fixes for Python 3.9

### PR DESCRIPTION
**Pull request by Gobot1234**
`type(field_type)` can now be `<class 'list'>` breaking the old method of checking.

### CODE REVIEW

File: src/betterproto/__init__.py [hunk 1]
```diff
@@ -818,9 +818,10 @@ def to_dict(
         """
         output: Dict[str, Any] = {}
         field_types = self._type_hints()
-        for field_name, meta in self._betterproto.meta_by_field_name.items():
+        proto_meta = self._betterproto
```
Comment (nat-n) [inline src/betterproto/__init__.py:821]: I don't think this it worth it here. I'm guessing the point is that for the cost of adding an assignment in scope, you save a lookup step for the value later?

However the perf value of this here is certainly below the level of noise for even a contrived test case, and I feel it makes the code less clear by adding a level of indirection without really clarifying anything in the process.

**Comment by nat-n** [comment]
Looks like the new CI config doesn't work?

**Comment by Gobot1234** [comment]
Yeah I'm not too sure what's up with it

File: src/betterproto/__init__.py (hunk 1 above)
Comment (Gobot1234) [reply src/betterproto/__init__.py:821]: Ok

**Comment by Gobot1234** [comment]
Looks like it's been fixed

**Comment by Gobot1234** [comment]
Not a clue what to do about this

**Comment by Gobot1234** [comment]
Can we cherry pick the actual fix to merged and leave the tests as a separate PR to be merged once 3.9 plays better?

**Comment by nat-n** [comment]
> Can we cherry pick the actual fix to merged and leave the tests as a separate PR to be merged once 3.9 plays better?

That would break the pipeline for other PRs.

**Comment by Gobot1234** [comment]
Building wheels for dependencies seems like an impossible task in actions.

File: src/betterproto/__init__.py [hunk 2]
```diff
@@ -926,9 +926,10 @@ def to_dict(
         """
         output: Dict[str, Any] = {}
         field_types = self._type_hints()
+        defaults = self._betterproto.default_gen
         for field_name, meta in self._betterproto.meta_by_field_name.items():
             field_type = field_types[field_name]
```
Comment (nat-n) [inline src/betterproto/__init__.py:931]: doesn't look like this variable has good reason to be here anymore.

**Review by nat-n — APPROVED** [review]
Why is the lockfile changed here?

**Comment by Gobot1234** [comment]
It was due to me experimenting with checking system platform specific dependencies.

File: src/betterproto/__init__.py (hunk 2 above)
Comment (Gobot1234) [reply src/betterproto/__init__.py:931]: I thought it would be better here to be consistent with `field_types`

**Comment by Gobot1234** [comment]
```py
>>> import typing
>>> def func(t: typing.List[str]):
...     pass
... 
>>> typing.get_type_hints(func)["t"]
typing.List[str]
>>> type(typing.get_type_hints(func)["t"])
<class 'typing._GenericAlias'>
>>> type(typing.List)
<class 'typing._SpecialGenericAlias'>
>>> type(typing.List) is type(typing.get_type_hints(func)["t"])
False
```
I think this is the actual bug.

File: src/betterproto/__init__.py (hunk 2 above)
Comment (nat-n) [reply src/betterproto/__init__.py:931]: Given the the value is no longer needed till near the end of the function, if at all, I think it's best to remove this assignment here now.

**Review by nat-n — CHANGES_REQUESTED** [review]
> Why is the lockfile changed here?

> It was due to me experimenting with checking system platform specific dependencies.

Better not to update the lockfile in this change if there's no good reason to do so.

File: src/betterproto/__init__.py (hunk 2 above)
Comment (Gobot1234) [reply src/betterproto/__init__.py:931]: It's used on https://github.com/Gobot1234/python-betterproto/blob/fixes/src/betterproto/__init__.py#L932 🤔

Comment (nat-n) [reply src/betterproto/__init__.py:931]: Just to make sure we're on the same page, I'm referring to `field_type`, not `defaults`.

Comment (Gobot1234) [reply src/betterproto/__init__.py:931]: I didn't add that

Comment (nat-n) [reply src/betterproto/__init__.py:931]: Indeed. Though you changed how it's used, so it would be good to clean it up in the process.

Comment (Gobot1234) [reply src/betterproto/__init__.py:931]: But wouldn't that mean the method was called multiple times in the loop when it could be called just once?

Comment (nat-n) [reply src/betterproto/__init__.py:931]: I'm talking about `field_type = field_types[field_name] ` which as it stands after you change is defined *inside* the loop and then maybe used once (though not used at all on most code paths).

**Review by nat-n — APPROVED** [review]

**Pull request merged**